### PR TITLE
476 nsa allow having a different configuration in the same json file for n and sa

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non regression tests were added (update of inputs, outputs, SA, algos)
- [x] main documentation was updated (update of input/output file)
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

**Comment by barondim** [comment]
- [ ] Documentation to be updated

**Comment by barondim** [comment]
Should I implement a NRT ?

File: etc/Dictionaries/DFLError_en_GB.dic [hunk 1]
```diff
@@ -12,6 +12,7 @@ ErrorConfigFileRead           =     error while reading configuration file: %1%
 StartingPointModeDoesntExist  =     starting point mode %1% doesn't exist
 InvalidActivePowerCompensation =    parameter ActivePowerCompensation can't be set on P in flat starting point mode in the input config file : %1%
 ChosenOutputDoesntExist       =     chosen output %1% doesn't exist
+NoFlatStartingPointModeInSA   =     can't set starting point mode on flat when simulation is in SA
```
Comment (rosiereflo) [inline etc/Dictionaries/DFLError_en_GB.dic:15]: I would propose "flat mode cannot be used in security analysis"

File: sources/Inputs/include/Configuration.h [hunk 2]
```diff
@@ -234,15 +238,19 @@ class Configuration {
   * @brief Helper function to update the starting point mode
   *
   * @param tree the element of the boost tree
+  * @param sa_mode true if simulation is in SA, false otherwise
   */
-  void updateStartingPointMode(const boost::property_tree::ptree& tree);
+  void updateStartingPointMode(const boost::property_tree::ptree& tree, const bool sa_mode);
```
Comment (rosiereflo) [inline sources/Inputs/include/Configuration.h:243]: coding style: saMode

File: sources/Inputs/src/Configuration.cpp [hunk 3]
```diff
@@ -34,14 +34,19 @@ namespace helper {
...
  */
 template<class T>
 static void
-updateValue(T& value, const boost::property_tree::ptree& tree, const std::string& key) {
-  auto value_opt = tree.get_child_optional(key);
-  if (value_opt.is_initialized()) {
+updateValue(T& value, const boost::property_tree::ptree& tree, const std::string& key, const bool sa_mode) {
+  std::string json_key = key;
+  if (sa_mode)
+    json_key = "sa." + key;
```
Comment (rosiereflo) [inline sources/Inputs/src/Configuration.cpp:44]: why not adding this directly in the key when calling update value?

File: sources/Inputs/src/Configuration.cpp [hunk 4]
```diff
@@ -150,35 +184,48 @@ Configuration::Configuration(const boost::filesystem::path& filepath, dfl::input
...
-    helper::updateValue(timeOfEvent_, config, "sa.TimeOfEvent");
-    helper::updateValue(timeStep_, config, "TimeStep");
-    helper::updateValue(tfoVoltageLevel_, config, "TfoVoltageLevel");
-    helper::updateActivePowerCompensationValue(activePowerCompensation_, config);
+    bool sa_mode = false;
+    if (request == dfl::common::Options::Request::RUN_SIMULATION_SA)
+      sa_mode = true;
+    updateStartingPointMode(config, sa_mode);
+    updateChosenOutput(config, simulationKind, sa_mode);
+    helper::updateValue(useInfiniteReactiveLimits_, config, "InfiniteReactiveLimits", sa_mode);
```
Comment (rosiereflo) [inline sources/Inputs/src/Configuration.cpp:192]: if only a N value is given then it should override the default for SA

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: sources/Inputs/include/Configuration.h [hunk 5]
```diff
@@ -43,9 +45,11 @@ class Configuration {
    *
    * @param filepath the configuration file to use
    * @param simulationKind the simulation kind (Steady state or Security analysis)
+   * @param request is the simulation in N or SA ?
```
[the remark this answers is not in the document]
Comment (barondim) [reply sources/Inputs/include/Configuration.h:48]: Test

File: documentation/configuringDynaflowLauncher/DynaflowLaunchersInputFiles.tex [hunk 6]
```diff
@@ -104,6 +104,8 @@ \subsection{Configuration file}
 
 The configuration file is a \textbf{.json} file containing the values for the parameters of Dynaflow-launcher. An example is given below:
 
+\subsubsection{Simple configuration}
+
```
Comment (barondim) [inline documentation/configuringDynaflowLauncher/DynaflowLaunchersInputFiles.tex:108]: Test

**Review by rosiereflo — APPROVED** [review]

**Review by marcochiaramello — APPROVED** [review]

**Comment by rosiereflo** [comment]
Could you please squash the commits?

**Pull request merged**